docs: add scrollBehavior API detail and link to guide (#954)
vuejs/vue-router-next: docs/api/index.md

### scrollBehavior

> Function to control scrolling when navigating between pages. Can return a ScrollPosition object or a Promise to delay scrolling.
>   
> See [Scroll Behaviour](../guide/advanced/scroll-behavior.md) for more details.

Reviewer: ```suggestion
Function to control scrolling when navigating between pages. Can return a Promise to delay when the scrolling happens. See [Scroll Behaviour](../guide/advanced/scroll-behavior.md) for more details.
```

### scrollBehavior

Function to control scrolling when navigating between pages. Can return:
  - a ScrollPosition object, which can contain:
    - [ScrollToOptions](https://developer.mozilla.org/en-US/docs/Web/API/ScrollToOptions)
    - an `el` property with a selector or reference to an element

>   - a Promise to delay scrolling

Reviewer: Can you revert this change? This is already included in the guide

Author: Just added a small bit of extra detail and removed the incomplete sentence.

function scrollBehavior(to, from, savedPosition) {
  // `to` and `from` are both route locations
  // `savedPosition` can be null if there isn't one

>   return savedPosition || { top: 0 };

Reviewer: this is just the API, let's remove this

```suggestion
```

Author: Fair enough, since the heading was "example" I thought it might be helpful to show a return value. Removing.
